app.py

```diff
@@ -115,119 +120,23 @@ def get_frame_range_from_shotgun(self):
-                               maxTime=out_frame,
-                               animationStartTime=in_frame,
-                               animationEndTime=out_frame)
-            
-            # set frame ranges for rendering
-            defaultRenderGlobals=pm.PyNode('defaultRenderGlobals')
-            defaultRenderGlobals.startFrame.set(in_frame)
-            defaultRenderGlobals.endFrame.set(out_frame)
-           
-        elif engine == "tk-nuke" and not self.engine.hiero_enabled:
-            import nuke
-
-            # unlock
-            locked = nuke.root()["lock_range"].value()
-            if locked:
-                nuke.root()["lock_range"].setValue(False)
-            # set values
-            nuke.root()["first_frame"].setValue(in_frame)
-            nuke.root()["last_frame"].setValue(out_frame)
-            # and lock again
-            if locked:
-                nuke.root()["lock_range"].setValue(True)
-
-        elif engine == "tk-motionbuilder":
-            from pyfbsdk import FBPlayerControl, FBTime
-
-            lPlayer = FBPlayerControl()
-            lPlayer.LoopStart = FBTime(0, 0, 0, in_frame)
-            lPlayer.LoopStop = FBTime(0, 0, 0, out_frame)
-
-        elif engine == "tk-softimage":
-            import win32com
-            Application = win32com.client.Dispatch('XSI.Application')
-            
-            # set playback control
-            Application.SetValue("PlayControl.In", in_frame)
-            Application.SetValue("PlayControl.Out", out_frame)
-            Application.SetValue("PlayControl.GlobalIn", in_frame)
-            Application.SetValue("PlayControl.GlobalOut", out_frame)       
-            
-            # set frame ranges for rendering
-            Application.SetValue("Passes.RenderOptions.FrameStart", in_frame)
-            Application.SetValue("Passes.RenderOptions.FrameEnd", out_frame)
-            
-
-        elif engine == "tk-houdini":
-            import hou
-            # We have to use hscript until SideFX gets around to implementing hou.setGlobalFrameRange()
-            hou.hscript("tset `((%s-1)/$FPS)` `(%s/$FPS)`" % (in_frame, out_frame))            
-            hou.playbar.setPlaybackRange(in_frame, out_frame)
-
-        elif engine == "tk-3dsmax":
-            from Py3dsMax import mxs
-            mxs.animationRange = mxs.interval(in_frame, out_frame)
-        elif engine == "tk-3dsmaxplus":
-            import MaxPlus 
-            ticks = MaxPlus.Core.EvalMAXScript("ticksperframe").GetInt()
-            range = MaxPlus.Interval(in_frame * ticks, out_frame * ticks)
-            MaxPlus.Animation.SetRange(range)
-        
-        else:
-            raise tank.TankError("Don't know how to set current frame range for engine %s!" % engine)
+        result = self.execute_hook_method(
+            "hook_frame_operation",
+            "set_frame_range",
+            in_frame=in_frame,
+            out_frame=out_frame
+        )
```

Comment:
Rather than have hooks catch and standardize the exception type, I think it's acceptable to let the hook raise an exception and have your exception catching logic here. This takes the onus off the hook developer, and lets you centralized your exception re-raising code.

Also, now that we're not returning a success bool, we can drop the return.

Maybe something like:
```
        try:
            self.execute_hook_method(
                "hook_frame_operation",
                "set_frame_range",
                in_frame=in_frame,
                out_frame=out_frame
            )
        except Exception as e:
            error_message = traceback.format_exc()
            self.log.error(error_message)
            raise tank.TankError("Encountered an error while setting the frame range: %s" % e)
```

Reply:
Done